pkg/apis/pipeline/v1/task_validation.go

```diff
@@ -300,6 +300,12 @@ func validateStep(ctx context.Context, s Step, names sets.String) (errs *apis.Fi
 			})
 		}
 	} else {
+		if len(s.Params) > 0 {
+			errs = errs.Also(&apis.FieldError{
+				Message: "params cannot be used without Ref",
+				Paths:   []string{"params"},
+			})
+		}
```

Comment:
please update the docs with this requirement

Reply:
done. PTAL `docs/stepactions.md`